### PR TITLE
Content-type dispatch via multimethod

### PR DESCRIPTION
**Pull request by plexus**
Duplicate of #548 to see if CircleCI is happier this way.

- [x] The commits are consistent with our [contribution guidelines](../blob/master/.github/CONTRIBUTING.md)
- [x] You've added tests to cover your change(s)
- [x] All tests are passing
- [x] The new code is not generating reflection warnings
- [x] You've updated the README (if adding/changing middleware)

**Note:** If you're just starting out to hack on `cider-nrepl` you might find
[nREPL's documentation](https://nrepl.org) and the
"Design" section of the README extremely useful.*

Thanks!

### CODE REVIEW

**Comment by plexus** [comment]
Hey, look at that @bbatsov !

**Comment by plexus** [comment]
Working on adding a test, then I think this is done.

**Comment by plexus** [comment]
This is good to go from my end! Happy that in the end all it took was three years of patience :D

**Comment by bbatsov** [comment]
@plexus Looks great! Just one small thing - you can't edit `ops.adoc` directly as it's auto-generated from the middleware descriptors (there's a `lein docs` task for this).

File: src/cider/nrepl/middleware/content_type.clj [hunk 1]
```diff
@@ -74,43 +75,44 @@
...
-
-    :else response))
+(defn external-body-response
+  "Partial response map having an external-body content-type referring to the given URL.
+
+  See RFC-2017: Definition of the URL MIME External-Body Access-Type."
+  [value]
+  {:content-type ["message/external-body"
+                  {"access-type" "URL"
+                   "URL" (.toString ^URL (as-url value))}]
```
Comment (vemv) [inline src/cider/nrepl/middleware/content_type.clj:85]: I think the type hint can be moved to the defprotocol

Comment (vemv) [inline src/cider/nrepl/middleware/content_type.clj:85]: It appears `(as-url)` is only ever invoked once, and immediately followed by a .toString?

If that's the case, could the URLCoercable protocol be string-based instead?

URLs are a legacy class, right? Especially in face of its infamous side-effecting `.equals` method.

**Comment by plexus** [comment]
Got it @bbatsov , I've made the doc changes in the middleware descriptors and ran `lein docs`.

File: src/cider/nrepl/middleware/content_type.clj (hunk 1 above)
Comment (plexus) [reply src/cider/nrepl/middleware/content_type.clj]: Thanks for the suggestion, I've changed `as-url` to `as-url-string`.

File: test/clj/cider/nrepl/middleware/content_type_test.clj [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+               {:URL (str "file:" f) :access-type "URL"}]
+              :status #{"done"}}
+             (-> {:op "eval"
+                  :code (str "(java.io.File. " (pr-str (str f)) ")")
+                  :content-type "true"}
+                 session/message
+                 (select-keys [:body :content-type :content-transfer-encoding :status]))))))
+
+  (testing "java.awt.image.RenderedImage"
+    (is (= {:body (if (SystemUtils/IS_JAVA_1_8)
```
[the remark this answers is not in the document]
Comment (plexus) [reply test/clj/cider/nrepl/middleware/content_type_test.clj:60]: I merely looked around to see how it was done elsewhere in the codebase. If this is overkill we can remove it here and in info-test in a separate PR.

**Review by vemv — APPROVED** [review]
Thanks for the iteration!

**Comment by bbatsov** [comment]
🥳 🚀

**Comment by plexus** [comment]
Yay! @arrdem must be happy we're finally out of his mentions :P

**Comment by arrdem** [comment]
Old friends always welcome 😉

**Comment by bbatsov** [comment]
@arrdem I thought we were all young friends. 😆

**Pull request merged**